src/scheduler.rs

```diff
@@ -526,17 +526,6 @@ async fn fetch_idle(
         }
     }
 
-    // Fetch the watched folder.
-    if let Err(err) = connection
-        .fetch_move_delete(ctx, &watch_folder, folder_meaning)
-        .await
-        .context("fetch_move_delete")
-    {
-        connection.trigger_reconnect(ctx);
-        warn!(ctx, "{:#}", err);
-        return InterruptInfo::new(false);
-    }
-
```

Comment:
Some lines further down, it says:

```
    // Scan additional folders only after finishing fetching the watched folder.
    //
    // On iOS the application has strictly limited time to work in background, so we may not
    // be able to scan all folders before time is up if there are many of them.
```

Now that `fetch_mode_delete` was moved down, it will be done after scanning, which will be a problem for iOS, then. Probably we need another `fetch_move_delete()` before the `loop`s. In `inbox_loop` we then need to make sure that `fetch_existing_msgs()` (and possibly `resync_folders()`, I didn't look into it) is executed _before_ `fetch_move_delete()`, which will need some additional logic.